### PR TITLE
Lazy load endpoints.

### PR DESCRIPTION
**Pull request by josejulio**
Lazy loading the endpoints. This will keep our users for having to do stuff like https://github.com/ManageIQ/manageiq/pull/13814

@cfcosta could you please review ?

// cc @pilhuhn

### CODE REVIEW

File: lib/hawkular/hawkular_client.rb [hunk 1]
```diff
@@ -47,13 +31,37 @@ def method_missing(name, *args, &block)
       delegate_client.__send__(method, *args, &block)
     end
 
+    def inventory
+      @inventory ||= Inventory::InventoryClient.create(entrypoint: "#{@state[:entrypoint]}/hawkular/inventory",
```
Comment (cfcosta) [inline lib/hawkular/hawkular_client.rb:35]: Why does `Metrics` has a different class name? Can we change the others to be like it? I like it better than to have `Inventory::Inventory` and `Alerts::Alerts`.

Comment (cfcosta) [reply lib/hawkular/hawkular_client.rb:35]: Also the initialization is different on `Inventory` but not on the others.

**Review by cfcosta — APPROVED** [review]
LGTM, there are some improvements that can be done but they might be out of scope of this PR.

File: lib/hawkular/hawkular_client.rb (hunk 1 above)
Comment (josejulio) [reply lib/hawkular/hawkular_client.rb:35]: I'm not sure, but at this point i don't want to break stuff like this: https://github.com/ManageIQ/manageiq/pull/13814/files#diff-7873cbd67d037fe54151ac4d277ccdd4R44

Comment (cfcosta) [reply lib/hawkular/hawkular_client.rb:35]: You can have this at the end of the file and keep it compatible:

```ruby
Inventory::InventoryClient = Inventory::Client
```

Comment (josejulio) [reply lib/hawkular/hawkular_client.rb:35]: Will open a separate PR to uniform it.

**Comment by josejulio** [comment]
@rubenvp8510 could you please merge?

**Pull request merged**